experimental/detect_breakage.py

```diff
@@ -0,0 +1,80 @@
+import enum
+import importlib
+import inspect
+import pkgutil
+
+def get_package_info(pkgname):
+    """returns a dict containing info about modules, classes, functions"""
+    pkg = importlib.import_module(pkgname)
```

Comment:
This means we should actually import each module we are going to analyze, and requires we install every dependencies of the whole package in order to import the modules. We should take care of the case when some of the dependencies were not installed by `pip install`, which will throw a `ModuleNotFoundError` exception.

e.g. `opencensus` has an extra dependency `google-cloud-trace` specified in `extras` field of the setup.py file.
```
pip install opencensus  # This won't install google-cloud-trace, and inspecting opencensus.trace.exporters will fail.

pip install opencensus[stackdriver] # This will install the extra deps, and inspect will succeed.
```

If we simply walk through the sub packages and inspect the modules, we will likely hit the case where the module dependencies are not installed. Probably we could do some error handling in that case.

Reply:
This also suggests we should run the breaking change detector in a virtual environment in order to not modify the packages installed in the user's environment. (If we want to install the extra deps in case they aren't installed).